src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -384,7 +380,7 @@ bool Creature::InitEntry(uint32 entry, CreatureData const* data /*= nullptr*/)
     SetSpeedRate(MOVE_SWIM,   1.0f); // using 1.0 rate
     SetSpeedRate(MOVE_FLIGHT, 1.0f); // using 1.0 rate
 
-    // Will set UNIT_FIELD_BOUNDINGRADIUS and UNIT_FIELD_COMBATREACH
+    // Will set UNIT_FIELD_BOUNDINGRADIUS, UNIT_FIELD_COMBATREACH and UNIT_FIELD_DISPLAYSCALE
     SetObjectScale(cinfo->scale);
```

Comment:
This field (OBJECT_SCALE_X) are always 1.0f now for creatures, the scale are done by UNIT_FIELD_DISPLAYSCALE and UNIT_FIELD_NATIVE_XDISPLAY_SCALE. You can have a value superior to 1.0f but it's done with an aura like mod size.